nicknamer.py

```diff
@@ -175,4 +177,65 @@ async def trace(context: Context) -> None:
         )
 
 
+@nicknamer.event
+async def on_message(message: Message) -> None:
+    cleaned_urls = {}
+
+    take_counter_measures, take_extreme_counter_measures = False, False
+
+    extractor = URLExtract()
+    for url in extractor.gen_urls(message.content):
+        clean_url = clear_url(url)
+
+        if url != clean_url:
+            cleaned_urls[url] = clean_url
+
+        take_counter_measures = take_counter_measures or len(clean_url) < len(url)
+        take_extreme_counter_measures = take_extreme_counter_measures or len(
```

Comment:
The equivalent condition for this is `filter(cleaned_urls.items(), lambda url, cleaned_url: len(clean_url) * URL_LENGTH_VIOLATION_FACTOR < len(url))`, so we can extract this variable after the loop.

Reply:
What's the point of doing that? It's not computationally more efficient or cleaner syntactically..

Reply:
Taking it out of the loop makes it clearer to read. I don't have to think about what are the states of the loop. Alternatively, you can add a function

```python
def is_critically_exceeded(url: str, cleaned_url: str) -> bool:
  return len(cleaned_url) * URL_LENGTH_VIOLATION_FACTOR < len(url)
```

and then use it!

Reply:
I mean I did it this way because I thought this was clearer to read haha, I'm thinking this is a style difference